File: src/main/java/com/clearcont/clearcontapp/helpers/Periodo.java
package com.clearcont.clearcontapp.helpers;

import org.jetbrains.annotations.NotNull;

import java.time.Month;
import java.time.format.TextStyle;
import java.util.Locale;

public class Periodo {
    static Locale locale = new Locale.Builder().setLanguage("pt").setRegion("BR").build();
    
    public static int getMonthByPeriodoString(@NotNull String periodo) {
        return switch (periodo) {
            case "JANEIRO" -> 1;
            case "FEVEREIRO" -> 2;
            case "MARÇO" -> 3;
            case "ABRIL" -> 4;
            case "MARIO" -> 5;
            case "JUNHO" -> 6;
            case "JULHO" -> 7;
            case "AGOSTO" -> 8;
            case "SETEMBRO" -> 9;
            case "OUTUBRO" -> 10;
            case "NOVEMBRO" -> 11;
            case "DEZEMBRO" -> 12;
            default -> throw new IllegalStateException("Unexpected value: " + periodo);
        };
    }
    
    public static @NotNull String getMonthByInt(Integer i) {
        return Month.of(i).getDisplayName(TextStyle.FULL, locale);
    }
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/helpers/Periodo.kt
package br.com.clearcont.clearcontwebapp.helpers

import java.time.Month
import java.time.format.TextStyle
import java.util.*

object Periodo {
    private var locale: Locale = Locale.Builder().setLanguage("pt").setRegion("BR").build()

    fun getMonthByPeriodoString(periodo: String): Int {
        return when (periodo) {
            "JANEIRO" -> 1
            "FEVEREIRO" -> 2
            "MARÇO" -> 3
            "ABRIL" -> 4
            "MARIO" -> 5
            "JUNHO" -> 6
            "JULHO" -> 7
            "AGOSTO" -> 8
            "SETEMBRO" -> 9
            "OUTUBRO" -> 10
            "NOVEMBRO" -> 11
            "DEZEMBRO" -> 12
            else -> throw IllegalStateException("Unexpected value: $periodo")
        }
    }

    fun getPortugueseMonthName(month: Month): String {
        return when (month) {
            Month.JANUARY -> "Janeiro"
            Month.FEBRUARY -> "Fevereiro"
            Month.MARCH -> "Março"
            Month.APRIL -> "Abril"
            Month.MAY -> "Maio"
            Month.JUNE -> "Junho"
            Month.JULY -> "Julho"
            Month.AUGUST -> "Agosto"
            Month.SEPTEMBER -> "Setembro"
            Month.OCTOBER -> "Outubro"
            Month.NOVEMBER -> "Novembro"
            Month.DECEMBER -> "Dezembro"
        }
    }

    fun getMonthByInt(i: Int?): String {
        return Month.of(i!!).getDisplayName(TextStyle.FULL, locale)
    }
}
